docs/docs/how_to/debugger/debugging_with_the_repl.md

```diff
@@ -0,0 +1,175 @@
+
+- Nargo: > x
+- Noir: > y
+- vscode-noir: > z
+
+## Debugging a simple circuit
+
+Let's debug a simple circuit:
+
+```rust
+fn main(x : Field, y : pub Field) {
+    assert(x != y);
+}
+```
+
+To start the REPL debugger, using a terminal, go to a Noir circuit's home directory. Then:
+
+`$ nargo debug`
+
+You should be seeing this in your terminal:
+
+```
+[main] Starting debugger
+At ~/noir-examples/recursion/circuits/main/src/main.nr:1:9
+  1 -> fn main(x : Field, y : pub Field) {
+  2        assert(x != y);
+  3    }
+> 
+```
+
+The debugger displays the current Noir code location, and it is now waiting for us to drive it.
+
+Let's first take a look at the available commands. For that we'll use the `help` command.
+
+```
+> help
+Available commands:
+
+  opcodes                          display ACIR opcodes
+  into                             step into to the next opcode
+  next                             step until a new source location is reached
+  out                              step until a new source location is reached
+                                   and the current stack frame is finished
+  break LOCATION:OpcodeLocation    add a breakpoint at an opcode location
+  over                             step until a new source location is reached
+                                   without diving into function calls
+  restart                          restart the debugging session
+  delete LOCATION:OpcodeLocation   delete breakpoint at an opcode location
+  witness                          show witness map
+  witness index:u32                display a single witness from the witness map
+  witness index:u32 value:String   update a witness with the given value
+  memset index:usize value:String  update a Brillig memory cell with the given
+                                   value
+  continue                         continue execution until the end of the
+                                   program
+  vars                             show variable values available at this point
+                                   in execution
+  stacktrace                       display the current stack trace
+  memory                           show Brillig memory (valid when executing a
+                                   Brillig block)
+  registers                        show Brillig registers (valid when executing
+                                   a Brillig block)
+  regset index:usize value:String  update a Brillig register with the given
+                                   value
+  step                             step to the next ACIR opcode
+
+Other commands:
+
+  help  Show this help message
+  quit  Quit repl
+
+```
+
+The command menu is for the most part self-explanatory. Some commands operate only at Brillig level, such as `memory`, `memset`, `registers`, `regset`. If you try to use them while execution is paused at an ACIR opcode, the debugger will simply inform you that you are not executing Brillig code:
```

Comment:
```suggestion
Some commands operate only for unconstrained functions, such as `memory`, `memset`, `registers`, `regset`. If you try to use them while execution is paused at an ACIR opcode, the debugger will simply inform you that you are not executing unconstrained code:
```
As much as I like the naming, I believe we're dropping "Brillig" in favour of the more self-explanatory name "unconstrained"

Reply:
I'll drop it from the docs explanations, but what's your opinion on the command descriptions show by the REPL itself?

For example:

`memory                           show Brillig memory (valid when executing a Brillig block)`

Some commands of the debugger are exposing the BrilligVM's internals, so the vocabulary for those will be kind of tricky to get right (it's one thing to talk about unconstrained code when explaining from Noir's perspective, but the semantic category of things like memory or registers doesn't match).